### PR TITLE
[rust-toolchain] Update to rust 1.61

### PR DESCRIPTION
**Pull request by gregnazario**
## Motivation

Upgrade rust to 1.61

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

CI

### CODE REVIEW

File: state-sync/state-sync-v1/src/fuzzing.rs [hunk 1]
```diff
@@ -10,13 +10,12 @@ use crate::{
     shared_components::test_utils,
 };
 use aptos_config::network_id::NetworkId;
-use aptos_infallible::Mutex;
```
Comment (JoshLind) [inline state-sync/state-sync-v1/src/fuzzing.rs:13]: What's the reason for removing this? 🤔

Comment (gregnazario) [reply state-sync/state-sync-v1/src/fuzzing.rs:13]: As per the clippy warnings, must use async aware mutex or it may be held on to across an await

Comment (JoshLind) [reply state-sync/state-sync-v1/src/fuzzing.rs:13]: Crazy

**Review by JoshLind — APPROVED** [review]
Thanks @gregnazario!

**Pull request closed without merging**